### ISSUE TITLE
Master Plan

### ISSUE THREAD

**Issue by jaanjah** [JaanJah/lua-game#1]
This issue is a master plan what MUST be completed before first release (v1.0)

Each step should be done as a separate Pull request (PR).
## Master Plan

- [ ] Add documentation
  - [x] README - What is this game, how to play, how to compile, https://github.com/JaanJah/base-defence/pull/7
  - [x] CONTRIBUTING - Code of conduct, https://github.com/JaanJah/base-defence/pull/7
  - [ ] CHANGELOG - Features in every release (Initial release in this case)
- [ ] Graphics
  - [ ] Sprites
    - [x] Player, https://github.com/JaanJah/base-defence/pull/3
    - [x] Enemies, https://github.com/JaanJah/base-defence/pull/5
    - [ ] Traps - Traps that trap enemies
    - [ ] Environment - Ground tiles, base sprites
    - [x] Spells, https://github.com/JaanJah/base-defence/pull/8
  - [x] Particles - OpenGL Shader Library?
- [ ] Gameplay
  - [x] Movement
    - [x] Player input, https://github.com/JaanJah/base-defence/pull/3
    - [x] Enemies movement, https://github.com/JaanJah/base-defence/pull/5
  - [ ] Map
    - [ ] Collision between objects
    - [x] Generate map, https://github.com/JaanJah/base-defence/pull/10
  - [x] Projectiles, https://github.com/JaanJah/base-defence/pull/8
  - [ ] UI
    - [ ] Score, Money
    - [ ] Interface for spells, traps
    - [x] Main menu, https://github.com/JaanJah/base-defence/pull/4
  - [ ] Spells
    - [x] Fireball, https://github.com/JaanJah/base-defence/pull/8
    - [ ] Ice wall
  - [ ] Traps
    - [ ] Freeze trap - when enemy walks on that tile, the enemy stops for 2 seconds.
- [ ] Code Quality
    - [ ] CI Pipeline
    - [x] [Editorconfig](https://editorconfig.org/)
    - [ ] Prettier
